### PR TITLE
test(NODE-4469): unskip all key management tests

### PR DESCRIPTION
**Pull request by baileympearson**
https://jira.mongodb.org/browse/NODE-4330

#### What is the motivation for this change?

This PR unskips all the key management API tests for the new key management API functions added in https://github.com/mongodb/libmongocrypt/pull/414.  

Before this PR merges, we need to

1. ~~merge https://github.com/mongodb/libmongocrypt/pull/414 into libmongocrypt~~
2. ~~update the supported commit hash in run-custom-csfle-test.sh to the new commit~~
3. ~~release another alpha of mongodb-client-encryption~~
4. ~~update to the newest version of mongodb-client-encryption for the regular tests in CI~~

### CODE REVIEW

**Review by addaleax — DISMISSED** [review]

File: test/tools/unified-spec-runner/unified-utils.ts [hunk 1]
```diff
@@ -110,17 +111,22 @@ export async function topologySatisfies(
...
+    const versionSupportsCSFLE = semverGte(config.version, '4.2.0');
+    const csfleEnabled = config.clientSideEncryption.enabled;
 
+    if (r.csfle) {
+      ok &&= versionSupportsCSFLE && csfleEnabled;
       if (!ok && skipReason == null) {
-        skipReason = `requires csfle to run but CSFLE is not set for this environment`;
+        skipReason = versionSupportsCSFLE
+          ? `requires csfle to run but CSFLE is not set for this environment`
+          : 'requires mongodb >= 4.2 to run csfle tests';
```
Comment (baileympearson) [inline test/tools/unified-spec-runner/unified-utils.ts:122]: Another fix to this logic - here's the actual spec https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#runonrequirement

The server must be >=4.2.0 AND csfle must be enabled for the driver.  For us, we rely on the configuration's `enabled` property to be consistent through all our tests.

File: .evergreen/run-custom-csfle-tests.sh [hunk 2]
```diff
@@ -27,7 +27,7 @@ ABS_PATH_TO_PATCH=$(pwd)
 # CSFLE_GIT_REF - set the git reference to checkout for a custom CSFLE version
 # CDRIVER_GIT_REF - set the git reference to checkout for a custom CDRIVER version (this is for libbson)
 
-CSFLE_GIT_REF=${CSFLE_GIT_REF:-e157c35ee4028b292883c4628dc5926f9742b34a}
+CSFLE_GIT_REF=${CSFLE_GIT_REF:-b14d06000535e55be0c619e936972805cac24d37}
```
Comment (baileympearson) [inline .evergreen/run-custom-csfle-tests.sh:30]: https://github.com/mongodb/libmongocrypt/commit/b14d06000535e55be0c619e936972805cac24d37 (latest mongodb-client-encryption release)

File: test/tools/unified-spec-runner/operations.ts [hunk 3]
```diff
@@ -510,7 +510,7 @@ operations.set('getKey', async ({ entities, operation }) => {
 operations.set('getKeys', async ({ entities, operation }) => {
   const clientEncryption = entities.getEntity('clientEncryption', operation.object);
 
-  return clientEncryption.getKeys();
+  return clientEncryption.getKeys().toArray();
```
Comment (baileympearson) [inline test/tools/unified-spec-runner/operations.ts:513]: `getKeys` returns a cursor, missed in the initial implementation of these operations in the unified runner

**Review by addaleax — APPROVED** [review]
lgtm!

**Review by nbbeeken — APPROVED** [review]

**Review by dariakp — CHANGES_REQUESTED** [review]
Just an organizational question here - was this supposed to be linked to a different subtask? The ticket this is linked to has already been closed, so I'm kind of confused where this sits with respect to the parent ticket

**Review by dariakp — APPROVED** [review]

**Pull request merged**